octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -240,14 +377,37 @@ async def get_symbol_prices(self,
                                 **kwargs: dict) -> typing.Optional[list]:
         try:
             with self.error_describer():
-                if limit:
+                if limit := self.cut_candle_limit(limit):
                     return await self.client.fetch_ohlcv(symbol, time_frame.value, limit=limit,
                                                          since=since, params=kwargs)
                 return await self.client.fetch_ohlcv(symbol, time_frame.value, since=since, params=kwargs)
         except ccxt.NotSupported:
-            raise octobot_trading.errors.NotSupported
+            if limit:
```

Comment:
Why should we only raise when limit is set?

Reply:
It's not clear that you want to execute the "retry with a limit"  code by passing the exception.

Reply:
I think you should create a method that tries to fetch with a limit and call it each time you catch an exception if not limit

Reply:
I moved it to a separate method

Reply:
:+1: